src/main/scala/com/bsgenerator/crawler/extractor/ExtractorCoordinator.scala

```diff
@@ -1,13 +1,108 @@
 package com.bsgenerator.crawler.extractor
 
-import akka.actor.{Actor, ActorLogging, Props}
+import akka.actor.{Actor, ActorRef, Props}
+import com.bsgenerator.crawler.CrawlingSupervisor
+import com.bsgenerator.utils.Helpers
 
 object ExtractorCoordinator {
   def props: Props = Props(new ExtractorCoordinator)
 
-  final case class Extract(content: String)
+  //  Messages:
+  final case class Extract(content: String, baseUrl: String)
+
+  final case class ExtractedContentAndLinks(requestId: String, content: Option[String], links: Set[String])
+
+  final case class FilteredLinks(requestId: String, links: Set[String])
+
 }
 
-class ExtractorCoordinator extends Actor with ActorLogging {
-  override def receive: Receive = Actor.emptyBehavior
+class ExtractorCoordinator extends Actor {
+
+  import ExtractorCoordinator._
+
+  protected val extractorsRouter: ActorRef = context.actorOf(ExtractorsRouter.props())
+
+  // For now, there is only one. In case it's bottleneck, create router for it as well
+  protected val storeManager: ActorRef = context.actorOf(StoreManager.props)
+
+
+  override def receive: Receive = waitForMessage(Set.empty, Set.empty, Set.empty)
+
+  def waitForMessage(extractRequests: Set[String],
+                     filterRequests: Set[String],
+                     storeLinksRequests: Set[String]): Receive = {
```

Comment:
extractRequests, filterRequests, storeLinksRequests sound like actions not collections. Maybe something along "extractingRequests" would be a better choice.